kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableFeaturesSuite.scala

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults
+
+import scala.collection.immutable.Seq
+
+import io.delta.kernel.engine.Engine
+import io.delta.kernel.expressions.Literal
+
+import org.apache.spark.sql.delta.DeltaLog
+import org.apache.spark.sql.delta.actions.Protocol
+
+/**
+ * Integration test suite for Delta table features.
+ */
+class DeltaTableFeaturesSuite extends DeltaTableWriteSuiteBase {
+
+  ///////////////////////////////////////////////////////////////////////////
```

Comment:
Do we have tests for things like if I try to enable CDF we will not be able to write?

Reply:
Added the unittests in TableFeaturesSuite.scala for each feature (supported, not enabled) and (supported, enabled) cases.

I haven't added the integration tests. I feel the unittests are good enough to cover.

Reply:
yeah those are very thorough! Can we add 1 test for 1 feature that's enabled by metadata but not supported by Kernel just to check the E2E integration & path through TransactionBuilderImpl? Like just this CDF enable case will fail